### PR TITLE
CR-1138133 xbutil advanced --write-mem --fill shows help menu on success

### PR DESCRIPTION
**Pull request by rchane**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1138133
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
The issue was first brought up in https://jira.xilinx.com/browse/CR-1136750 before this CR was created. 
#### How problem was solved, alternative solutions (if any) and why they were rejected
`xbutil advanced --write-mem --fill` now properly returns after finishing successfully. The help menu's redundant `size` arg was removed (non-required arguments will not be printed beforehand if they are positional arguments). In addition, a bug where `--write-mem`'s help menu returned a non-zero exit code has been fixed.
#### Risks (if any) associated the changes in the commit
This affects the display of the "usage" information in the help menus for commands.
#### What has been tested and how, request additional testing if necessary
Testing was done on vck5000.
```
rchane@xsjsagarw50:/proj/rdi/staff/rchane/XRT$ xbutil advanced --write-mem --help

DESCRIPTION: Write to a given memory address

USAGE: xbutil advanced --write-mem [--help] [-i arg] [-f arg] [--count arg] -d arg address size
  <address>       - Base address to start from
  <size>          - Block size (bytes) to write

OPTIONS:
  -i, --input        - Input file
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  --count            - Number of blocks to write
  -f, --fill         - The byte value to fill the memory with
  --help             - Help to use this sub-command


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
rchane@xsjsagarw50:/proj/rdi/staff/rchane/XRT$ xbutil advanced --write-mem --fill 0xaa -d 0000:65:00.1 0x0 1024
Memory write succeeded
```
#### Documentation impact (if any)
N/A

### CODE REVIEW

**Review by chvamshi-xilinx — COMMENTED** [review]
Hi @rchane , is there any reason for choosing address & size as positional arguments?

**Comment by rchane** [comment]
Hello @chvamshi-xilinx, address & size were positional arguments previously, so I left them as such!

**Comment by chvamshi-xilinx** [comment]
> Hello @chvamshi-xilinx, address & size were positional arguments previously, so I left them as such!

Ok. Thanks Ryan. This PR looks good if we are considering these as positional arguments.
Can you please consider converting these position arguments to regular arguments in your next PR as this is anyway undocumented feature.

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by rchane** [comment]
The commit CR number is incorrect, but the CR link/PR title has been updated to reflect the correct CR.

File: src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp [hunk 1]
```diff
@@ -205,6 +205,7 @@ OO_MemWrite::execute(const SubCmdOptions& _options) const
       addr += size;
     }
     std::cout << "Memory write succeeded" << std::endl;
+    return;
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp:208]: Why is there a need to add `return` here?

File: src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp [hunk 2]
```diff
@@ -58,16 +58,16 @@ OO_MemWrite::execute(const SubCmdOptions& _options) const
 {
   XBU::verbose("SubCommand option: write mem");
 
-  // Parse sub-command ...
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp:61]: Could you elaborate why the following changes were required.   These changes bind's this code to the "--help" option name and no longer relies on the "m_help" variable.

Looking over the comments it is mentioned that:
_In addition, a bug where --write-mem's help menu returned a non-zero exit code has been fixed._

I assume that this is the bug that has been addressed, but not 100% sure.

**Review by rozumx — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp (hunk 1 above)
Comment (rchane) [reply src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp:208]: Without `return`, the code previously went on after a successful write to `printHelp()` and throws an error.

File: src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp (hunk 2 above)
Comment (rchane) [reply src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp:61]: Yes, that is the bug! `process_arguments` was originally throwing an error when using `xbutil advanced --write-mem --help` due to the missing `--address` option. I then tried to make the help option check consistent with OO_MemRead.cpp. From my understanding, "m_help" is not set until `process_arguments` is called.

**Review by rozumx — APPROVED** [review]

**Pull request merged**